compute/backup/manage-backups.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python
+
+# pip install --upgrade google-api-python-client
+# pip install --upgrade iso8601
+# pip install --upgrade rfc3339
+# pip install --upgrade pytz
+
+import simplejson as json
+import iso8601
+import datetime
+import pytz
+import sys
+from oauth2client.client import GoogleCredentials
+from googleapiclient import discovery
+import re
+import argparse
+
+DISK_ZONE_MAP = {}
+
+def list_snapshots(compute, project, filter=None, pageToken=None):
+    result = compute.snapshots().list(project=project, pageToken=pageToken, filter=filter).execute()
+    return result
+
+# Given a list of snapshot items, return True if a snapshot should be
+# taken, False if not.
+def should_snapshot(items, minimum_delta):
+    _items = items[:]
+    _items.sort(key=lambda x: x['creationTimestamp'])
+    _items.reverse()
+    if datetime.datetime.now(pytz.utc) > iso8601.parse_date(_items[0]['creationTimestamp']) \
+        + minimum_delta:
+        return True
+    return False
+
+
+# Given a list of snapshot items, return the snapshots than can be
+# deleted.
+def deletable_items(items):
+    _items = items[:]
+    _items.sort(key=lambda x: x['creationTimestamp'])
+    _items.reverse()
+
+    result = []
+    now = datetime.datetime.now(pytz.utc)
+    one_week = datetime.timedelta(days=7)
+    three_months = datetime.timedelta(weeks=13)
+    one_year = datetime.timedelta(weeks=52)
+    minimum_number = 1
+
+    # Strategy: look for a reason not to delete. If none found,
+    # add to list.
+
+    # Global reasons
+
+    if len(items) < minimum_number:
+        print "Fewer than %d snapshots, not deleting any" % minimum_number
```

Comment:
Use `.format` instead of `%` and always use print as a function for Python 3 compatibility.